alternator/executor.cc

```diff
@@ -3391,6 +3392,10 @@ static future<executor::request_return_type> do_query(service::storage_proxy& pr
             // update our "filtered_row_matched_total" for all the rows matched, despited the filter
             cql_stats.filtered_rows_matched_total += items["Items"].Size();
         }
+        // TODO: better threshold
```

Comment:
Since at this point we already created the JSON, don't we have a better estimate of its size instead of the number of results?
Please note that with SELECT=COUNT or filtering, we may have 100 results but they are very small (with count - even nothing), so just the number of results may not be indicative of the size of the response.

Reply:
I considered "measuring" the object, but it seems potentially expensive to traverse and estimate. Though obviously the "heuristicts" currently used are pure bogus. I am open for suggestions.

Reply:
Yeah, I wanted to measure on something that did not add an obvious check overhead, and doing hashed lookup on the actual json element + check count would. But the check right now might be completely bogus indeed. The best option would probably be to make `describe_items` also return actual elem count/payload size.

Reply:
What I meant in my last comment was that even if you just want the item count (and not their sizes), better use `items["Items"].Size()` (as in two lines up) and not rs.size(), because the former is the number of results after filtering and the latter is before filtering.

Reply:
Yes, I understood that. I did not do this because of the overhead of `items["Items"]`, though negligible...

Reply:
Updated to use items.size() (and save a few name lookups)